src/python/pants/core/util_rules/system_binaries.py

```diff
@@ -529,5 +529,98 @@ async def find_tar_wrapper(_: TarBinaryRequest, tar_binary: TarBinary) -> TarBin
     return tar_binary
 
 
+# -------------------------------------------------------------------------------------------
+# Binary shims
+# Creates a Digest with a shim for each requested binary in a directory suitable for PATH.
+# -------------------------------------------------------------------------------------------
+
+
+@dataclass(frozen=True)
+class BinaryShimsRequest:
+    requests: tuple[BinaryPathRequest, ...]
+    rationale: str
```

Comment:
I think we want to turn off equality for this: 

https://github.com/pantsbuild/pants/blob/2a01ab7dbe9431fc7ec7cdb550a323804b748322/src/python/pants/engine/process.py#L51